File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        
        newSet = {}

        for idx in range(len(nums)):
            if nums[idx] in newSet and abs(idx - newSet[nums[idx]]) <= k:
                return True
            newSet[nums[idx]] = idx

        return False
                

File: 219-contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> last = new HashMap<>();//create a new hashmap.
        for(int i = 0; i < nums.length; i++){//run a forloop through the array.
            Integer j = last.put(nums[i], i);//set Integer j to map last.
            if(j != null && i-j <= k){//check the given conditions.
                return true;//if correct, return true.
            }
        }
        return false;//otherwise return false.
    }
}